.circleci/config.yml

```
@@ -27,15 +27,39 @@ references:
     run:
       name: Install dependencies
       command: |
-        # Since we're testing against a very old cmake, wheels may not exist
-        # for Python versions >=3.8. We will need to build the cmake wheels
-        # from a tarball in this case, and cmake depends on scikit-build, so we
-        # install skbuild separately here. Note that newer cmake versions build
-        # universal wheels instead, so once we upgrade to cmake >=3.14 we
-        # won't be dependent on them updating wheels for specific new Python
-        # versions.
-        ${PYTHON} -m pip install --user --progress-bar off scikit-build==0.10.0
-        ${PYTHON} -m pip install --user --progress-bar off -U -r requirements-testing.txt
+        # Create a virtual environment to ignore the CI image's pre-installed packages
+        ${PYTHON} -m pip install --progress-bar off --upgrade virtualenv --user
+        mkdir -p ./venv
+        ${PYTHON} -m virtualenv ./venv --clear
+        . venv/bin/activate
+        ${PYTHON} -m pip install --progress-bar off --upgrade pip virtualenv
```

Why are we upgrading `virtualenv` again?

---

That might have been an oversight on my part. I'll change it to only upgrade pip.
```suggestion
        ${PYTHON} -m pip install --progress-bar off --upgrade pip
```